libnavui-status/src/test/java/com/mapbox/navigation/ui/status/view/MapboxStatusViewTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.mapbox.navigation.ui.status.view
+
+import android.content.Context
+import android.os.Build
+import androidx.core.view.isVisible
+import androidx.test.core.app.ApplicationProvider
+import com.mapbox.navigation.ui.status.R
+import com.mapbox.navigation.ui.status.model.Status
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@Config(sdk = [Build.VERSION_CODES.O_MR1])
+@RunWith(RobolectricTestRunner::class)
+class MapboxStatusViewTest {
```

Comment:
Can we add a test to check the behavior of `cancel` and thus `hide`

@tomaszrybakiewicz

Reply:
I don't think we will be able to do that. For some reason the onAnimationEnd and onAnimationCancel on hideAnimator listener is not being called when testing with Robolectric.

I tried using coroutines to wait for `isRendered` value update before verifying it but with no luck. Second coroutine times out even with 5sec timeout:

```kotlin
    @Test
    fun `cancel - should hide the view`() = runBlocking {
        val status = Status("message", animated = false, icon = android.R.drawable.ic_secure)
        sut.render(status)
        awaitUntil { sut.isRendered }

        sut.cancel()
        awaitUntil(5000) { !sut.isRendered }
        assertFalse("isVisible", sut.isRendered)

    }

    private suspend fun awaitUntil(timeout: Long = 1000, predicate: () -> Boolean) = withTimeout(timeout) {
        while(true) {
            if (predicate()) return@withTimeout
            delay(50)
        }
    }
```

Any ideas how to test hide anim?

Reply:
```
@Test
    fun `cancel - should hide the view`() {
        val status = Status("message", animated = true, icon = android.R.drawable.ic_secure)
        sut.render(status)
        sut.cancel()

        ShadowLooper.idleMainLooper()

        assertFalse("isVisible", sut.isRendered)
    }
```

Reply:
Nice. Thanks 🙏.